Competition/src/main/cpp/subsystems/Shooter.cpp

```diff
@@ -0,0 +1,185 @@
+    state.flywheelState = FlywheelState::FLYWHEEL_DEFAULT;
+}
+
+void Shooter::resetEncoder(){
+    turretEncoder.SetPosition(0);
+    hoodEncoder.SetPosition(0);
+}
+
+void Shooter::assessInputs()
+{
+    if (!operatorController)
+    {
+        return;
+    }
+
+    // driver inputs
+    state.leftStickX = operatorController->GetLeftX();
+    state.startButton = operatorController->GetStartButton();
+    state.backButton = operatorController->GetBackButton(); 
+    //state.dPadDownPressed = driverController->GetPOV(frc::GenericHID::)
+    
+    //Turret
+    if (std::abs(state.leftStickX) > ShooterConstants::kDeadband) {
+        state.turretState = TurretState::TURRET_MANUAL;
+    }
+    else if(state.startButton){
+       state.turretState = TurretState::TURRET_PRIME;
+    }
+    else if(state.backButton){
+        state.turretState = TurretState::TURRET_DEFAULT;
+    }
+
+    //Hood
+    if(state.startButton){
+        state.hoodState = HoodState::HOOD_PRIME;
+    }
+    else if(state.backButton){
+        state.hoodState = HoodState::HOOD_DISABLE;
+    }
+
+    //Flywheel
+    if(state.startButton){
+        state.flywheelState = FlywheelState::FLYWHEEL_PRIME;
+    }
+    else if (state.backButton){
+        state.flywheelState = FlywheelState::FLYWHEEL_DEFAULT;
+    }
+}
+
+void Shooter::analyzeDashboard()
+{
+    if(table->GetBoolean("Home Turret", false)){
+        state.turretState = TurretState::TURRET_HOME;
+    }
+}
+
+void Shooter::assignOutputs()
+{   
+    state.turretTarget = 0;
+
+    //DISABLED
+    if(state.turretState == TurretState::TURRET_DISABLE){
+        state.turretTarget = 0;
+    }//HOME
+    else if(state.turretState == TurretState::TURRET_HOME){
+        // @TODO Motion Profile to turret home position
+        state.turretTarget = 0;
+    } //MANUAL
+     else if (state.turretState == TurretState::TURRET_MANUAL) {
+        int sign = state.leftStickX >= 0 ? 1 : -1;
+        state.turretTarget = sign * std::pow(state.leftStickX, 2) * ShooterConstants::TURRET_SPEED_MULTIPLIER;
+
+        // Minimum power deadband
+        if (std::abs(state.turretTarget) < ShooterConstants::pDeadband) {
+            state.turretTarget = 0;
+        }
+        // Stop deadband
+        else if (std::abs(state.turretTarget) < ShooterConstants::pSoftDeadband) {
+            int direction = 1;
+            if (state.turretTarget < 0) direction = -1;
+            state.turretTarget = ShooterConstants::pSoftDeadband * direction;
+        }
+    }//DEFAULT
+    else if (state.turretState == TurretState::TURRET_DEFAULT){
+        //Odometry tracking
+        // @TODO Motion Profile to target turret position
+        state.turretTarget = 0;
+    }
+
+    turret.Set(state.turretTarget);
+
+    if(state.flywheelState == FlywheelState::FLYWHEEL_DISABLE){
+        state.flywheelTarget = 0;
+    }
+    else if(state.flywheelTarget == FlywheelState::FLYWHEEL_PRIME){
+        //math conditions, leaving 1 for now
+        // @TODO use dashboard value
```

Comment:
Value should be changed via a dashboard item

Reply:
Additionally need to use falcon control mode for velocity